src/server/scripts/EasternKingdoms/Deadmines/instance_deadmines.cpp

```diff
@@ -91,9 +100,12 @@ class instance_deadmines : public InstanceMapScript
                 {
                     case CANNON_GUNPOWDER_USED:
                         CannonBlast_Timer = DATA_CANNON_BLAST_TIMER;
-                        // it's a hack - Mr. Smite should do that but his too far away
-                        //pIronCladDoor->SetName("Mr. Smite");
-                        //pIronCladDoor->MonsterYell(SAY_MR_SMITE_ALARM1, LANG_UNIVERSAL, NULL);
+
+                        // Loads Mr. Smite's grid. We need this as he must be active in order for his texts to work.
+                        instance->LoadGrid(MrSmite.GetPositionX(), MrSmite.GetPositionY());
```

Comment:
incorrect

if (Creature* smite = instance->GetCreature(MrSmiteGUID))
{
    Position pos = smite->GetPosition();
    instance->LoadGrid(pos.GetPositionX(), pos.GetPositionY());
}
But i think loading grid here is not required, try changing textRange in `creature_text`
Ranges:
0 - Normal (Default)
1 - Area
2 - Zone
3 - Map
4 - World